backend/plugins/utilities/textfsm/template.py

```diff
@@ -133,27 +115,45 @@ def insert_template_into_index_lines(self, original_template_lines: typing.List[
 
             elif driver_section_identified and count == 0:  # first line after the last in the right driver section
                 count += 1
-                new_line = f'{template_filename}, .*, {driver}, {command}\n'
+                new_line = f"{template_filename}, .*, {driver}, {command}\n"
                 new_index_lines.append(new_line)
 
             new_index_lines.append(line)
 
         return new_index_lines
 
 
+def return_errors(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        try:
+            result_data = f(*args, **kwargs)
+        except Exception as e:
+            result_data = {
+                "status": "error",
+                "data": {"error": str(e)}
+            }
+        return result_data
+
+    return wrapper
```

Comment:
I suspect we'll end up using this elsewhere, and probably migrate this to some better home.  A `utils` module or something somewhere.  But it's here for now

Reply:
This *could* be applied to the class methods themselves.  applying it only to the root level functions makes it easier to test the methods though.  It's easier to test something that cleanly breaks and throws an exception than something that might or might not have an error code buried in it somewhere.